### PR TITLE
CY-1374 Python 3 support

### PR DESCRIPTION
**Pull request by tehasdf**
- [x]  apply 2to3 fix-by-fix and tune the outputs maually
- [x] run unittests on py3
- [x] make flake pass on py3 (should be immediate after 2to3 is done)
- [x] make the tests pass on py2 again
- [ ] make the tests pass on py3

### CODE REVIEW

**Comment by tehasdf** [comment]
2to3 fixes:
- [ ] apply
- [ ] asserts
- [ ] basestring
- [ ] buffer
- [ ] callable
- [ ] dict
- [ ] except
- [ ] exec
- [ ] execfile
- [ ] exitfunc
- [ ] filter
- [ ] funcattrs
- [ ] future
- [ ] getcwdu
- [ ] has_key
- [ ] idioms
- [ ] import
- [ ] imports
- [ ] imports2
- [ ] input
- [ ] intern
- [ ] isinstance
- [ ] itertools
- [ ] itertools_imports
- [ ] long
- [ ] map
- [ ] metaclass
- [ ] methodattrs
- [ ] ne
- [ ] next
- [ ] nonzero
- [ ] numliterals
- [ ] operator
- [ ] paren
- [ ] print
- [ ] raise
- [ ] raw_input
- [ ] reduce
- [ ] renames
- [ ] repr
- [ ] set_literal
- [ ] standarderror
- [ ] sys_exc
- [ ] throw
- [ ] tuple_params
- [ ] types
- [ ] unicode
- [ ] urllib
- [ ] ws_comma
- [ ] xrange
- [ ] xreadlines
- [ ] zip

File: dsl_parser/import_resolver/abstract_import_resolver.py [hunk 1]
```diff
@@ -58,10 +54,10 @@ def read_import(import_url):
     error_str = 'Import failed: Unable to open import url'
     if import_url.startswith('file:'):
         try:
-            request = urllib2.Request(import_url)
-            with contextlib.closing(urllib2.urlopen(request)) as f:
```
Comment (tehasdf) [inline dsl_parser/import_resolver/abstract_import_resolver.py]: this is under a `if startswith('file:')`, wtf :D

**Comment by ptlm** [comment]
@tehasdf is there a plan forward with this PR? There are many vital changes made here missing from the PR merged in September. On the other hand, a few conflicts were introduced in master.

Would you rather continue this branch, or perhaps split it into a couple smaller and more reviewable PRs?

**Comment by tehasdf** [comment]
@ptlm indeed this PR will most likely become a few smaller ones instead; we made this one, as a draft, to see the extent of changes needed; however, then it turned out that python 3 support is not in scope for the upcoming release of Cloudify, and it'll be part of the next (5.1) release instead, hopefully

that of course will require more changes than just the ones in this PR (because there's several more repositories, as you know), and will start to happen - I'm not the one who's prioritizing tasks, though - I would imagine in January

**Pull request closed without merging**